src/client.js

```diff
@@ -224,7 +224,7 @@ Client.prototype.connect = function(args) {
 
 	if (config.lockNetwork) {
 		// This check is needed to prevent invalid user configurations
-		if (args.host && args.host.length > 0 && args.host !== config.defaults.host) {
+		if (!Helper.config.public && args.host && args.host.length > 0 && args.host !== config.defaults.host) {
```

Comment:
Servers can come from user configuration only in private mode, so in public mode we can just enforce it.